src/cluster.c

```diff
@@ -1527,7 +1527,12 @@ static void linkHandleResponse(redisAsyncContext *c, void *r, void *privdata)
     JoinLinkState *state = ConnGetPrivateData(conn);
 
     if (!reply) {
+#ifndef HAVE_TLS
         LOG_ERROR("RAFT.SHARDGROUP GET failed: connection dropped.");
+#else
+        unsigned long e = ERR_peek_last_error();
+        LOG_ERROR("RAFT.SHARDGROUP GET failed: connection dropped: %s", ERR_reason_error_string(e));
```

Comment:
Consider keeping these ifdefs local and have a single function to handle getting the extended error information?

Besides that, are we sure we even need to report an SSL error always? For example, if we experience a TCP connect error - wouldn't we report a stale and misleading OpenSSL error here?

Reply:
re: last point, this seems like it should be hiredis then?

Reply:
so I now understand the first part.  you dislike the #ifdef/else/#endif block in this location, and would rather a connectionFailedReasong() that produces a string and would contain it.

with that said, my comment on the last point still stands, as all connection handling is abstracted away from hiredis users, hiredis itself would have to tell us why the connection failed to do it in any way reliably, and I just don't see how to do that at the moment.

Reply:
adding 2 tickets for redisraft and hiredis https://github.com/RedisLabs/redisraft/issues/296